sequoia/http.py

```diff
@@ -92,20 +100,29 @@ def return_response(response, resource_name):
 
     def request(self, method, url, data=None, params=None, headers=None, retry_count=0, resource_name=None):
 
+        def http_status_codes_to_retry():
+            retry_codes = self.backoff_strategy.pop(RETRY_HTTP_STATUS_CODES, [])
+            retry_codes = list(map(int, retry_codes)) if isinstance(retry_codes, (list, tuple)) else [int(retry_codes)]
+            retry_codes.append(429)
+            return retry_codes
+
+        retry_http_status_codes = http_status_codes_to_retry()
+
         def fatal_code(e):
-            return isinstance(e, error.HttpError) and \
-                   400 <= e.status_code < 500 and e.status_code != 429 \
-                   or isinstance(e, error.AuthorisationError)
+            return isinstance(e, error.AuthorisationError) or \
+                   isinstance(e, error.HttpError) and \
+                   400 <= e.status_code < 500 and \
+                   e.status_code not in retry_http_status_codes
 
-        def backoff_hdlr(details):
-            logging.warning('Retry `%s` for args `%s` and kwargs `%s`', details['tries'], details['args'],
-                            details['kwargs'])
+        def enable_backoff_logs():
+            logging.getLogger('backoff').addHandler(logging.StreamHandler())
 
+        enable_backoff_logs()
         decorated_request = backoff.on_exception(self.backoff_strategy.pop('wait_gen', backoff.constant),
                                                  (error.ConnectionError, error.Timeout, error.TooManyRedirects,
                                                   error.HttpError),
                                                  giveup=fatal_code,
-                                                 on_backoff=backoff_hdlr,
+                                                 max_time=self.backoff_strategy.pop('max_time', self.MAX_TIME_SECONDS),
```

Comment:
if you set the max_time with the default value as i comment before, i think we don't need these line.